utils/s3/client.cc

```diff
@@ -82,6 +81,19 @@ future<> ignore_reply(const http::reply& rep, input_stream<char>&& in_) {
     co_await util::skip_entire_stream(in);
 }
 
+future<> look_for_errors(const http::reply&, input_stream<char>&& in_) {
+    auto in = std::move(in_);
+    auto body = co_await util::read_entire_stream_contiguous(in);
+    auto possible_error = aws::aws_error::parse(std::move(body));
+    if (possible_error.get_error_type() != aws::aws_error_type::OK) {
+        if (possible_error.is_retryable()) {
+            // Pretend we have networking problem to force the client to retry
+            throw std::system_error(std::error_code(EPIPE, std::generic_category()));
```

Comment:
Client will retry only once if seeing this. More correct handling should be done externally. The handler function needs to fail the request with some "generic" exception, then the caller (finalize_upload()) re-sends the request again as many times as the exception is retriable

Reply:
it will be addressed (if there are no objections) in another PR (WiP) [here](https://github.com/kreuzerkrieg/scylladb/tree/EWZ/even-more-aws-errors-and-retry)

Reply:
I'd prefer keeping only the part that handles "inner" (encoded in XML) error and aborts (and tune-up the retryable test respectively). The retry can be done as followup if it's complicated

Reply:
Addressed